frontend/src/utilities/useCustomK8sWatchResource.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  K8sModelCommon,
+  K8sResourceCommon,
+  WatchK8sResource,
+  WebSocketOptions,
+  useK8sWatchResource,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import React from 'react';
+import { CustomWatchK8sResult } from '~/types';
+
+const useCustomK8sWatchResource = <T extends K8sResourceCommon | K8sResourceCommon[]>(
+  initResource: WatchK8sResource | null,
+  initModel?: K8sModelCommon,
+  options?: Partial<WebSocketOptions & RequestInit & { wsPrefix?: string; pathPrefix?: string }>,
+): CustomWatchK8sResult<T> => {
+  const [data, loaded, error] = useK8sWatchResource<T>(initResource, initModel, options);
+
+  const loadError = React.useMemo(() => {
+    if (error instanceof Error || typeof error === 'undefined') {
+      return error;
+    }
+
+    if (typeof error === 'string') {
+      return undefined;
+    }
```

Comment:
TBH i don't think the SDK has any case where it's not `Error` or `undefined`. But if it does, I would expect if `error` is defined, we should be returning an instance of `Error`.

Reply:
`useK8sWatchResource` is returning an empty string on success.  How will we handle that.  `[data, true,'']`

Reply:
Ok in that case, do something like this:
```
    if (error instanceof Error) {
      return error;
    }

    if (!error) {
      return undefined;
    }
```